infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationMetadataManager.java

```diff
@@ -335,17 +338,146 @@ public void setSnapshotAppliedComplete(LogReplicationEntry entry) {
+                    .setSnapshotSyncInfo(currentSyncInfo)
+                    .build();
+
+            corfuStore.tx(NAMESPACE)
+                    .update(REPLICATION_STATUS_TABLE, key, current, null)
+                    .commit();
+
+            log.debug("updateSnapshotSyncInfo as completed: clusterId: {}, syncInfo: {}",
+                    clusterId, currentSyncInfo);
+        }
+    }
+
+    /**
+     * Set replication status table.
+     * If the current sync type is log entry sync, keep Snapshot Sync Info.
+     *
+     * @param clusterId standby cluster id
+     * @param remainingEntries num of remaining entries to send
+     * @param type sync type
+     */
+    public void setReplicationStatusTable(String clusterId, long remainingEntries,
+                                          ReplicationStatusVal.SyncType type) {
+        ReplicationStatusKey key = ReplicationStatusKey.newBuilder().setClusterId(clusterId).build();
+        LogReplicationMetadata.SnapshotSyncInfo syncInfo = null;
+        ReplicationStatusVal current;
+
+        CorfuRecord<ReplicationStatusVal, Message> record =
+                corfuStore.query(NAMESPACE).getRecord(REPLICATION_STATUS_TABLE, key);
+        if (record != null) {
+            ReplicationStatusVal previous = record.getPayload();
+            syncInfo = previous.getSnapshotSyncInfo();
+        }
+
+        if (type == ReplicationStatusVal.SyncType.LOG_ENTRY) {
+            if (syncInfo == null){
+                log.warn("setReplicationStatusTable during LOG_ENTRY sync, " +
+                        "previous status is not present for cluster: {}", clusterId);
+                syncInfo = LogReplicationMetadata.SnapshotSyncInfo.newBuilder().build();
+            }
+
+            current = ReplicationStatusVal.newBuilder()
+                    .setRemainingEntriesToSend(remainingEntries)
+                    .setSyncType(type)
+                    .setStatus(LogReplicationMetadata.SyncStatus.ONGOING)
+                    .setSnapshotSyncInfo(syncInfo)
+                    .build();
+
+            corfuStore.tx(NAMESPACE)
+                    .update(REPLICATION_STATUS_TABLE, key, current, null)
+                    .commit();
+
+            log.debug("setReplicationStatusTable: clusterId: {}, remainingEntries: {}, type: {}, syncInfo: {}",
+                    clusterId, remainingEntries, type, syncInfo);
+        } else if (type == ReplicationStatusVal.SyncType.SNAPSHOT) {
+            LogReplicationMetadata.SnapshotSyncInfo currentSyncInfo;
+            if (syncInfo == null){
+                log.warn("setReplicationStatusTable during SNAPSHOT sync, " +
+                        "previous status is not present for cluster: {}", clusterId);
+                currentSyncInfo = LogReplicationMetadata.SnapshotSyncInfo.newBuilder().build();
+            } else {
+                currentSyncInfo = syncInfo.toBuilder()
+                        .setStatus(LogReplicationMetadata.SyncStatus.ONGOING)
```

Comment:
if we reach this and the snapshotSyncInfo is not present, shouldn't we set it to NOT_STARTED? as the path once it is started will set it to ONGOING... how do we handle potential TAE's from here and from the other async updateSnapshotSyncInfo? cause we could then overwrite a ONGOING, with a NOT_STARTED...

Reply:
If we reach here, current sync type is snapshot sync. So I have to keep both statuses as ongoing...
All replication status updates are protected by the lock in the ackReader, so TAE should be fine right now.
I added a flag to avoid overwriting by the periodic task.

Reply:
BTW, I haven't found a way to differentiate NOT_STARTED and STOPPED if the current statue is initialized. All update is based on replication FSM, but negotiation is a runtime state.